File: implementations/python/sdk/src/multiplexed_ai_sdk/contracts/replay/execution_replay_response.py
from dataclasses import dataclass, field

from ...wire_model import AiSdkWireModel
from ..common.schema_versions import AiSdkSchemaVersions


@dataclass(frozen=True)
class AiSdkExecutionReplayResponse(AiSdkWireModel):
    execution_id: str
    succeeded: bool
    started_at_utc: str
    completed_at_utc: str
    duration_ms: int
    schema_version: int = AiSdkSchemaVersions.EXECUTION_REPLAY_RESPONSE
    deterministic: bool | None = None
    message: str | None = None
    diagnostics: tuple[str, ...] = field(default_factory=tuple)
    failure_reason: str | None = None
    correlation_id: str | None = None

File: implementations/node/sdk/src/contracts/replay/execution-replay-response.ts
import { AI_SDK_SCHEMA_VERSIONS } from "../common/schema-versions.js";

export interface AiSdkExecutionReplayResponse {
  readonly schemaVersion: typeof AI_SDK_SCHEMA_VERSIONS.executionReplayResponse;
  readonly executionId: string;
  readonly succeeded: boolean;
  readonly deterministic?: boolean;
  readonly message?: string;
  readonly diagnostics: readonly string[];
  readonly failureReason?: string;
  readonly correlationId?: string;
  readonly startedAtUtc: string;
  readonly completedAtUtc: string;
  readonly durationMs: number;
}
